sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1434,7 +1434,9 @@ class DataFrame private[sql](
     }
   }
 
-  private[sql] def internalRowRdd = queryExecution.executedPlan.execute()
+  // Note that the RDD is memoized. Once called, it won't change even if you change any
+  // query planning related Spark SQL configurations.
+  private[sql] def internalRowRdd = queryExecution.toRdd
```

Comment:
`queryExecution.toRdd` is already a lazy val, why are we creating another another name for that at all?